pkg/smart/ataidentify.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2018 The OpenEBS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package smart
+
+import (
+	"fmt"
+
+	"github.com/openebs/node-disk-manager/pkg/util"
+)
+
+// swapByteOrder swaps the order of every second byte in a byte slice and modifies the slice in place.
+func (d *IdentDevData) swapByteOrder(b []byte) []byte {
```

Comment:
What is IdentDevData()? An SCSI page that we got a return value, that we are parsing now?  I think so, right?

So as we move one we will add more pages that get added to this, where several mode/log pages and perhaps others as well, will tell us more and more about the disk as we go. So, should we not reflect that we are parsing pages? (byte arrays)? i.e

func( p *page) whatever() (what, ever)  {

}

We know by inspecting the page what it contains..

Reply:
IdentDevData here is the name of the struct which we are using to parse the page defined by ATA8-ACS as a command set for ATA devices.
Right @gila . It would be good to name it something as "ATACSPageStruct", abbreviated for ATA command set Page struct.